save extended properties only when asked for

AmitPhulera opened the pull request:
Fixes https://github.com/celery/django-celery-results/issues/315

Currently `django-celery-results` stores all the info about the task to DB whereas the default behavior as per celery should only be storing some basic properties. Extended properties should be saved only when they are requested for. It can be done by setting `extended-result` flag in the celery config.

The PR checks for that flag while storing the results and proceeds accordingly.

I would be adding some tests for it once we are okay with the approach.

auvipy reviewed (changes requested):
this will need unit tests to cover

AmitPhulera commented:
@auvipy Thanks for the quick turnaround. Yeah, I am planning to add unit tests for it. Just wanted to be sure that this is the right way to go forward.

AmitPhulera commented:
Hey @auvipy, 
I have added a test to verify the flag works https://github.com/celery/django-celery-results/pull/316/commits/7edbd42250f751b654108fa55e09ba714d4b6efd and also updated previous tests to use the flag [6974a76](https://github.com/celery/django-celery-results/pull/316/commits/6974a76a02152cb2bbe13edc2957c400d67f35ee)

AmitPhulera commented:
Hey @auvipy, 
Could you please let me know what would be the next steps in this PR?

auvipy commented:
@AllexVeldman can you check this please

AmitPhulera commented on django_celery_results/backends/database.py:
```diff
@@ -54,6 +54,53 @@ def exception_safe_to_retry(self, exc):
+            'task_kwargs': None,
+            'task_name': None,
+            'traceback': None,
+            'using': None,
```
https://github.com/celery/django-celery-results/pull/316/commits/bde3eefc74bc6cd91b860bb032184c6b7bb0b247

AmitPhulera commented on django_celery_results/backends/database.py:
```diff
@@ -54,6 +54,53 @@ def exception_safe_to_retry(self, exc):
                 return True
         return False
 
+    def _get_extended_properties(self, request, traceback, using):
```
https://github.com/celery/django-celery-results/pull/316/commits/36aeaebff3046713c6923c2d185028b903acae9d

AllexVeldman replied:
Hi @AmitPhulera, I think I was not entirely clear.
My intention was for you to call the `self._get_result_meta` defined by the super class, as it already implements conditionally populating the meta based on `self.app.conf.find_value_for_key('extended', 'result')`

With that meta you can adjust it to the django_celery_results backend specifics like `task_args` instead of `args`, adding `periodic_task_name` if needed and storing argsrepr and kwargsrepr if specified.

AmitPhulera replied:
Hey @AllexVeldman, 
I tried to use `self._get_result_meta()` but I don't think it was of any use. We have to alter/remove many properties and then we have to use `self.app.conf.find_value_for_key('extended', 'result')` anyways to add `peridoic_task_name`.
I felt like we were taking the worst of both worlds it did not simplify anything. But this is just my opinion if you feel that approach would be better then I can push the changes and make it use `self._get_result_meta()`.

AllexVeldman replied:
makes sense, then I prefer your original method name over overriding `_get_result_meta`, as its returned properties are different.

AmitPhulera replied:
Hey @AllexVeldman, 
I have reverted the commits to use `_get_extended_properties `.

AmitPhulera commented:
Thanks for your review @AllexVeldman, I have addressed the feedback that you have shared. Let me know if there is anything else that you want me to update.

auvipy reviewed (changes requested):
should we document the changes?

AmitPhulera commented:
@auvipy Can you please guide me on what should be the correct place to document it? 
The [celery docs](https://docs.celeryq.dev/en/stable/userguide/configuration.html#result-extended) already have information about the same. 
Do you think an entry in changelog would suffice?

auvipy commented:
If you can please dig the docs and find out any suitable place to add, otherwise an entry to changelog would be alright. but I would request to check the docs more initially.

AmitPhulera commented:
@auvipy  I was actually looking into the docs earlier to figure out if this can be communicated in a better way. Still, the only docs that I came across were [these](https://django-celery-results.readthedocs.io/en/latest/) which mention the installation(which points to celery docs), API, and changelog.

I think only entry in changelog makes sense here

If you have any other place in mind, I would be happy to make the changes there.

auvipy commented:
ok put in change log please

AmitPhulera commented:
@auvipy I have added the changes in Changelog. See https://github.com/celery/django-celery-results/pull/316/commits/af09b8784cebb1014dd7a0b924168a84dfab1bf1

I have put it in the minor version bump rather than updating the patch version. Feel free to update as you seem fit.

AmitPhulera commented:
Hey @auvipy, 
Would you be able to provide an estimate on when a release tag would be created for the change?

auvipy commented:
new update https://github.com/celery/django-celery-results/issues/326#issuecomment-1177478454

AmitPhulera commented:
I guess the change is okay, right?

valberg commented:
Could we add a FAQ or something to the docs to inform of this? Maybe even copy the "Getting started" from https://docs.celeryq.dev/en/latest/django/first-steps-with-django.html#django-celery-results-using-the-django-orm-cache-as-a-result-backend and include a note in that?

auvipy commented:
please go ahead

valberg commented:
Here you go :) https://github.com/celery/django-celery-results/pull/328

G-Rath commented:
Looks like this resolves #142 aka GHSA-fvx8-v524-8579 🎉

auvipy commented:
wow
